### PR TITLE
Update state of developoment in README

### LINKED ISSUE #2578: Outdated version info in the readme
Now that SFML 2.6.0 is out, these 2 sentences need to be updated:

https://github.com/SFML/SFML/blob/7af8719a4929928c7f169472f16a9ee2894f2863/readme.md?plain=1#L12-L13

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Closes #2578 

I removed mention of the 2.6.x branch since now that 2.6.0 is out, there is relatively little reason to use 2.6.x or know that it exists. It has been relegated to a boring implementation detail. I'd rather users check out the 2.6.0 tag.

### CODE REVIEW

**Review by eXpl0it3r — APPROVED** [review]

**Comment by ChrisThrasher** [comment]
Fixed a stale link and reworded some things now that the state of development is actually not so hard to follow anymore.

**Review by eXpl0it3r — APPROVED** [review]
